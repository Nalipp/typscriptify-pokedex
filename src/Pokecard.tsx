import "./Pokecard.css";
import { IPokemon } from "./interfaces";

const POKE_API = "https://raw.githubusercontent.com/" +
    "PokeAPI/sprites/master/sprites/pokemon/";


/** Individual Pokemon card.
 *
 * Props:
 * - title
 * - image
 * - type
 * - base_experience: (number of experience points)
 *
 * */

    function Pokecard({ base_experience, id, name, type }: IPokemon) {
  const imgSrc = `${POKE_API}${id}.png`;

  return (
      <div className="Pokecard">
        <div className="Pokecard-title">{name}</div>
        <img className="Pokecard-image" src={imgSrc} alt={name} />
        <div className="Pokecard-data">Type: {type}</div>
        <div className="Pokecard-data">EXP: {base_experience}</div>
      </div>
  );
}

export default Pokecard;
