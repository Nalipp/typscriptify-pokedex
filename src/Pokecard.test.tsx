import { render } from '@testing-library/react';
import Pokecard from './Pokecard';

test('Pokecard component will render with props', () => {
  render(
    <Pokecard 
       base_experience={62} 
       id={4} name={'Charmander'} 
       type={'fire'} />
    );
});

test('Pokecard component matches snapshot', () => {
  const { container } = render(
    <Pokecard 
       base_experience={62} 
       id={4} name={'Charmander'} 
       type={'fire'} />
    );

  expect(container).toMatchSnapshot();
})


