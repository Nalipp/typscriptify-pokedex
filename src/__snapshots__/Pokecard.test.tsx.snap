// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokecard component matches snapshot 1`] = `
<div>
  <div
    class="Pokecard"
  >
    <div
      class="Pokecard-title"
    >
      Charmander
    </div>
    <img
      alt="Charmander"
      class="Pokecard-image"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
    />
    <div
      class="Pokecard-data"
    >
      Type: 
      fire
    </div>
    <div
      class="Pokecard-data"
    >
      EXP: 
      62
    </div>
  </div>
</div>
`;
